### ISSUE TITLE
poetry shell fails after not finding bash in virtualenv

### ISSUE THREAD

**Issue by DrPyser** [sdispater/poetry#638]
<!--
  Hi there! Thank you for discovering and submitting an issue.

  Before you submit this; let's make sure of a few things.
  Please make sure the following boxes are ticked if they are correct.
  If not, please try and fulfill these first.
-->

<!-- Checked checkbox should look like this: [x] -->
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

<!--
  Once those are done, if you're able to fill in the following list with your information,
  it'd be very helpful to whoever handles the issue.
-->

- **OS version and name**: archlinux
- **Poetry version**: 0.12.8

## Issue
<!-- Now feel free to write your issue, but please be descriptive! Thanks again 🙌 ❤️ -->
Running `poetry shell` fails with:
```
$ poetry shell -vvv
Using virtualenv: /home/charles/.cache/pypoetry/virtualenvs/kafka-utils-py3.7
Spawning shell within /home/charles/.cache/pypoetry/virtualenvs/kafka-utils-py3.7
                                                                                                                  
[FileNotFoundError]                                                                                
[Errno 2] No such file or directory: '/home/charles/.cache/pypoetry/virtualenvs/kafka-utils-py3.7/bin/bash': '/home/charles/.cache/pypoetry/virtualenvs/kafka-utils-py3.7/bin/bash'                                                                                     
                                                                                                                  
Exception trace:
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /home/charles/.poetry/lib/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /home/charles/.poetry/lib/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /home/charles/.poetry/lib/poetry/console/commands/shell.py in handle() at line 41
   self.env.execute(shell.path)
 /home/charles/.poetry/lib/poetry/utils/env.py in execute() at line 480
   return super(VirtualEnv, self).execute(bin, *args, **kwargs)
 /home/charles/.poetry/lib/poetry/utils/env.py in execute() at line 295
   return subprocess.call([bin] + list(args), **kwargs)
 /usr/lib/python3.7/subprocess.py in call() at line 317
   with Popen(*popenargs, **kwargs) as p:
 /usr/lib/python3.7/subprocess.py in __init__() at line 769
   restore_signals, start_new_session)
 /usr/lib/python3.7/subprocess.py in _execute_child() at line 1516
   raise child_exception_type(errno_num, err_msg, err_filename)

shell
```

**Comment by DrPyser** [comment]
And using `poetry run` fails in the same way for any command not in virtualenv. I guess the `PATH` isn't set correctly.

**Comment by utkugenc** [comment]
I have the same problem on Windows 10 with both git-bash (mintty) and Windows cmd (ConEmu):

```bash
$ poetry --version
Poetry 0.12.8
```

git-bash (`SHELL=/usr/bin/bash`):
```
$ poetry shell -vvv
Using virtualenv: C:\Users\ugenc\AppData\Local\pypoetry\Cache\virtualenvs\poetry-test-py3.6
Spawning shell within C:\Users\ugenc\AppData\Local\pypoetry\Cache\virtualenvs\poetry-test-py3.6

[RuntimeError]
Unable to detect the current shell.

Exception trace:
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 C:\Users\ugenc\.poetry\lib\poetry\console\application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 C:\Users\ugenc\.poetry\lib\poetry\console\commands\command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\commands\base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\commands\command.py in execute() at line 107
   return self.handle()
 C:\Users\ugenc\.poetry\lib\poetry\console\commands\shell.py in handle() at line 40
   shell = Shell.get()
 C:\Users\ugenc\.poetry\lib\poetry\utils\shell.py in get() at line 37
   raise RuntimeError("Unable to detect the current shell.")

shell
```

Windows cmd (`SHELL` is not defined) :
```
$ poetry shell -vvv
Using virtualenv: C:\Users\ugenc\AppData\Local\pypoetry\Cache\virtualenvs\poetry-test-py3.6
Spawning shell within C:\Users\ugenc\AppData\Local\pypoetry\Cache\virtualenvs\poetry-test-py3.6

[FileNotFoundError]
[WinError 2] The system cannot find the file specified

Exception trace:
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 C:\Users\ugenc\.poetry\lib\poetry\console\application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 C:\Users\ugenc\.poetry\lib\poetry\console\commands\command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\commands\base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 C:\Users\ugenc\.poetry\lib\poetry\_vendor\py3.6\cleo\commands\command.py in execute() at line 107
   return self.handle()
 C:\Users\ugenc\.poetry\lib\poetry\console\commands\shell.py in handle() at line 41
   self.env.execute(shell.path)
 C:\Users\ugenc\.poetry\lib\poetry\utils\env.py in execute() at line 480
   return super(VirtualEnv, self).execute(bin, *args, **kwargs)
 C:\Users\ugenc\.poetry\lib\poetry\utils\env.py in execute() at line 295
   return subprocess.call([bin] + list(args), **kwargs)
 C:\Users\ugenc\AppData\Local\Programs\Python\Python36\lib\subprocess.py in call() at line 267
   with Popen(*popenargs, **kwargs) as p:
 C:\Users\ugenc\AppData\Local\Programs\Python\Python36\lib\subprocess.py in __init__() at line 709
   restore_signals, start_new_session)
 C:\Users\ugenc\AppData\Local\Programs\Python\Python36\lib\subprocess.py in _execute_child() at line 997
   startupinfo)

shell
```

**Comment by sdispater** [comment]
It should be fixed in the latest `0.12.9` release.
